pugjs/types.go

```diff
@@ -328,6 +334,19 @@ func (a *Array) Pop() Object {
 	return last
 }
 
+// Shift removes an element from the beginning of the array and returns the element
+func (a *Array) Shift() Object {
+	first := a.items[0]
+	a.items = a.items[1:len(a.items)]
+	return first
+}
+
+// Unshift adds an element to the beginning of the array and returns the new length
+func (a *Array) Unshift(elements ...Object) int {
```

Comment:
`Push` would be the equivalent function: this allows to add a single item returns `Object`. But `Unshift` allows to add vararg and returns the new length.

Could this be improved so that both functions have a closer behavior?

Reply:
The idea about this function is, that the behavior is as close as possible to the JS implementation of the function. So having the same input and return values then in JS